python-package/xgboost/sklearn.py

```diff
@@ -556,7 +592,7 @@ def load_model(self, fname):
 
     def _configure_fit(
         self,
-        booster: Optional[Booster],
+        booster: Optional[Booster, "XGBModel"],
```

Comment:
This line appears to cause the trouble. What is this for? Can we revert it?

Reply:
This is for the `xgb_model` parameter in `fit` method.  It can be both a `Booster` object and a sklearn wrapper, which should inherit from `XGBModel`.  I will fix that.

Reply:
Sorry for asking you to look into that,  I meant the jenkins CI, it seems you have restarted it.

Reply:
In that case, you should use `Optional[Union[Booster, "XGBModel"]]`

Reply:
Done.

Reply:
@trivialfis According to https://stackoverflow.com/questions/43022843/nvidia-nvml-driver-library-version-mismatch, rebooting fixes this error:
```
Failed to initialize NVML: Driver/library version mismatch
```
Apparently, the EC2 instances performed an unattended (automatic) upgrades. All I did was to kill all worker instances and restart the failed job. Now the problem is fixed.

Reply:
@hcho3 I think maybe, some workers are still lingering around?  https://xgboost-ci.net/blue/organizations/jenkins/xgboost/detail/PR-6591/14/pipeline

Reply:
No, I shut them down.